### ISSUE TITLE
 an illegal memory access was encountered

### ISSUE THREAD

**Issue by GavChap** [comfyanonymous/ComfyUI_TensorRT#9]
I'm getting an error on using the static TRT with height 768 width 1344 with an SDXL checkpoint.

Error occurred when executing SamplerCustom:

CUDA error: an illegal memory access was encountered
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

**Comment by aikitoria** [comment]
Same here.

**Comment by comfyanonymous** [comment]
context_min, opt, etc is the multiplier, so the min amount would be context_min*77 and the max amount would be context_max*77 since for SD models the tokens are in chunks of 77.

**Comment by aikitoria** [comment]
Ohh, that makes more sense. So with 128 I was just setting it to nonsense. Let me try again!

**Comment by aikitoria** [comment]
Yes, with a more sane context size, it works great for me now :)

I'm not seeing a significant difference in execution time to my previous dynamic engine with variable context length, but the static one seems to use less VRAM.

**Comment by jkrauss82** [comment]
Not sure if it is the same problem but neither the static nor the dynamic TRT work.

I am trying to convert an SDXL model on an RTX 4060 Ti with 16G VRAM.

![image](https://github.com/comfyanonymous/ComfyUI_TensorRT/assets/3026023/c929da32-90d3-4645-86b6-7732339f069a)

**env:**
```
$ cat /proc/version
Linux version 6.5.0-25-generic (buildd@bos03-amd64-044) (x86_64-linux-gnu-gcc-12 (Ubuntu 12.3.0-1ubuntu1~22.04) 12.3.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #25~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Feb 20 16:09:15 UTC 2

$ python -V
Python 3.11.0rc1

$ nvidia-smi
Wed Jun  5 15:45:57 2024
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 545.23.06              Driver Version: 545.23.06    CUDA Version: 12.3     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 4060 Ti     On  | 00000000:01:00.0  On |                  N/A |
|  0%   48C    P8              18W / 165W |    394MiB / 16380MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+

ComfyUI:
Total VRAM 16081 MB, total RAM 31952 MB
pytorch version: 2.3.0+cu121
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 4060 Ti : cudaMallocAsync
VAE dtype: torch.bfloat16
Using pytorch cross attention
```

**comfyui log:**

```
got prompt
model_type EPS
Using pytorch attention in VAE
Using pytorch attention in VAE
Requested to load SDXL
Loading 1 new model
/home/workspace/ComfyUI/comfy/ldm/modules/diffusionmodules/openaimodel.py:846: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
assert y.shape[0] == x.shape[0]
/home/workspace/ComfyUI/comfy/ldm/modules/diffusionmodules/openaimodel.py:123: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
assert x.shape[1] == self.channels
/home/workspace/ComfyUI/comfy/ldm/modules/diffusionmodules/openaimodel.py:81: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
assert x.shape[1] == self.channels
[06/05/2024-15:29:41] [TRT] [I] The logger passed into createInferBuilder differs from one already provided for an existing builder, runtime, or refitter. Uses of the global logger, returned by nvinfer1::getLogger(), will return the existing value.
[06/05/2024-15:29:41] [TRT] [I] [MemUsageChange] Init CUDA: CPU +0, GPU +0, now: CPU 0, GPU 696 (MiB)
[06/05/2024-15:29:45] [TRT] [I] [MemUsageChange] Init builder kernel library: CPU +0, GPU +312, now: CPU 0, GPU 1008 (MiB)
[06/05/2024-15:29:45] [TRT] [I] ----------------------------------------------------------------
[06/05/2024-15:29:45] [TRT] [I] Input filename:   /home/workspace/ComfyUI/temp/1717594112.6238039/model.onnx
[06/05/2024-15:29:45] [TRT] [I] ONNX IR version:  0.0.8
[06/05/2024-15:29:45] [TRT] [I] Opset version:    17
[06/05/2024-15:29:45] [TRT] [I] Producer name:    pytorch
[06/05/2024-15:29:45] [TRT] [I] Producer version: 2.3.0
[06/05/2024-15:29:45] [TRT] [I] Domain:
[06/05/2024-15:29:45] [TRT] [I] Model version:    0
[06/05/2024-15:29:45] [TRT] [I] Doc string:
[06/05/2024-15:29:45] [TRT] [I] ----------------------------------------------------------------
No timing cache found; Initializing a new one.
Building engine:  50%|████████████████████████████████████████████████████████████████████▌                                                                    | 3/6 [00:00<00:00,  3.88it/s][06/05/2024-15:29:56] [TRT] [I] Global timing cache in use. Profiling results in this builder pass will be stored.                                                     | 0/1 [00:00<?, ?it/s]
 0%|                                                                                                                                                  | 0/3 [00:00<?, ?it/s]
Timing graph nodes:  47%|████████████████████████████████████████████████████████████▋                                                                     | 118/253 [03:46<04:41,  2.08s/it]
./start.sh: line 5: 3878990 Killed                  PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libtcmalloc_minimal.so.4 python main.py --listen 0.0.0.0 --port 7860
```

**Note:**

I have tried with python 310 and with disabling tcmalloc and expandable segments, same error.

Let me know if I can help by providing more logs etc.

**Comment by aikitoria** [comment]
Supporting up to batch size 6 at a resolution up to 1344x1344 is probably too big for that GPU. Try building separate engines for vertical/horizontal resolutions and using smaller batch size.

**Comment by jkrauss82** [comment]
I should have written that I have tried different settings (for the static model I have just used batch size 1) and the error is always that the process abort around step 120 in stage `timing graph nodes`.

I have monitored the VRAM, it does not go above ~1.5G at any stage/step. I think it is most likely something else, however I do not see any other error message unfortunately, not even the dreaded `segmentation fault`.

**Comment by gabrie** [comment]
No matter how I build the engine, I tried batch opt=2/4/8, then when I run them, it always crashes on the second time queue prompt with the same error. Sometime even it crashes on the first time.
